compiler/rustc_target/src/asm/sparc.rs

```diff
@@ -0,0 +1,134 @@
+use std::fmt;
+
+use rustc_data_structures::fx::FxIndexSet;
+use rustc_span::Symbol;
+
+use super::{InlineAsmArch, InlineAsmType, ModifierInfo};
+use crate::spec::{RelocModel, Target};
+
+def_reg_class! {
+    Sparc SparcInlineAsmRegClass {
+        reg,
+        yreg,
+    }
+}
+
+impl SparcInlineAsmRegClass {
+    pub fn valid_modifiers(self, _arch: super::InlineAsmArch) -> &'static [char] {
+        &[]
+    }
+
+    pub fn suggest_class(self, _arch: InlineAsmArch, _ty: InlineAsmType) -> Option<Self> {
+        None
+    }
+
+    pub fn suggest_modifier(
+        self,
+        _arch: InlineAsmArch,
+        _ty: InlineAsmType,
+    ) -> Option<ModifierInfo> {
+        None
+    }
+
+    pub fn default_modifier(self, _arch: InlineAsmArch) -> Option<ModifierInfo> {
+        None
+    }
+
+    pub fn supported_types(
+        self,
+        arch: InlineAsmArch,
+    ) -> &'static [(InlineAsmType, Option<Symbol>)] {
+        match self {
+            Self::reg => {
+                // FIXME: i64 is ok for g*/o* registers on SPARC-V8+ ("h" constraint in GCC)
+                if arch == InlineAsmArch::Sparc {
+                    types! { _: I8, I16, I32; }
+                } else {
+                    types! { _: I8, I16, I32, I64; }
+                }
+            }
+            Self::yreg => &[],
+        }
+    }
+}
+
+pub(crate) fn is_v7_or_v8(arch: InlineAsmArch, target: &Target) -> bool {
+    // FIXME: There are any number of CPUs that imply SPARC-V8+, and checking the cpu
+    // string in the target specs is not the ideal approach here, but there
+    // doesn't seem to be another good way to do this yet (https://github.com/rust-lang/rust/pull/131222).
+    // However, in LLVM 20, feature "+v8plus" will be available for this:
+    // https://github.com/llvm/llvm-project/commit/aca971d336d9c7650120fc0fd6dfe58866408216
+    arch == InlineAsmArch::Sparc && target.options.cpu != "v9"
+}
```

Comment:
cc https://github.com/rust-lang/rust/pull/131222

Reply:
Perhaps adding the v9 target feature and referencing it (and after LLVM 20, adding and referencing the v8plus target feature instead of v9.) would be a good idea?
https://github.com/llvm/llvm-project/blob/llvmorg-19.1.0/llvm/lib/Target/Sparc/Sparc.td#L35

Reply:
Implemented the target feature based approach in https://github.com/rust-lang/rust/pull/132472/commits/14f6234fc81f9cf9eafbd0f4ea3773fd1ae71913.